Tests/RxFeedbackTests/FeedbackLoopsTests.swift

```diff
@@ -0,0 +1,212 @@
+        let effects: () -> Observable<String> = {
+            return .just("_a")
+        }
+        let feedback: (ObservableSchedulerContext<String>) -> Observable<String> = react(query: query, effects: effects)
+        let system = Observable.system(
+            initialState: "initial",
+            reduce: { oldState, append in
+                return  oldState + append
+            },
+            scheduler: scheduler,
+            scheduledFeedback: feedback
+        )
+        let expected = scheduler.createHotObservable([
+            next(201, "initial"),
+            next(204, "initial_a")
+            ])
+
+        // Run
+        let results = scheduler.start { system }
+
+        // Test
+        XCTAssertEqual(results.events, expected.recordedEvents)
+    }
+
+    func testFeedbacksOrder() {
+        // Prepare
+        let scheduler = TestScheduler(initialClock: 0)
+        let query1: (String) -> Void? = { state in
+            if state == "initial" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let query2: (String) -> Void? = { state in
+            if state == "initial_a" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let effects1: () -> Observable<String> = {
+            return .just("_a")
+        }
+        let effects2: () -> Observable<String> = {
+            return .just("_b")
+        }
+        let feedback1: (ObservableSchedulerContext<String>) -> Observable<String>
+        feedback1 = react(query: query1, effects: effects1)
+        let feedback2: (ObservableSchedulerContext<String>) -> Observable<String>
+        feedback2 = react(query: query2, effects: effects2)
+        let system = Observable.system(
+            initialState: "initial",
+            reduce: { oldState, append in
+                return  oldState + append
+            },
+            scheduler: scheduler,
+            scheduledFeedback: feedback1, feedback2
+        )
+        let expected = scheduler.createHotObservable([
+            next(201, "initial"),
+            next(204, "initial_a"),
+            next(206, "initial_a_b")
+            ])
+
+        // Run
+        let results = scheduler.start { system }
+
+        // Test
+        XCTAssertEqual(results.events, expected.recordedEvents)
+    }
+
+    func testFeedbacksCancelation() {
+        // Prepare
+        let scheduler = TestScheduler(initialClock: 0)
+        let notImediateEffect = PublishSubject<String>()
+        let query1: (String) -> Void? = { state in
+            if state == "initial" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let query2: (String) -> Void? = { state in
+            if state == "initial" {
+                return ()
+            } else {
+                return nil
+            }
+        }
+        let effects1: () -> Observable<String> = {
```

Comment:
This test would need to make sure `effects` lambda is being called at least once. Otherwise `effects` not being returned could be a result of `effects1` lambda not being called, and not a cancellation logic working properly.